packages/discovery-react-components/src/components/DocumentPreview/components/PdfViewerHighlight/PdfViewerHighlight.tsx

```diff
@@ -127,4 +162,33 @@ const useHighlighter = ({
   }, [document, isReady, pageNum, pdfRenderedText, processedDoc, textMappings]);
 };
 
+function useScrollIntoActiveHighlight(
+  highlightDivRef: React.MutableRefObject<HTMLDivElement | null>,
+  shapes: (HighlightShape | undefined)[],
+  activeIds: string[] | undefined
+) {
+  useEffect(() => {
+    if (!highlightDivRef.current) {
+      return;
+    }
+
+    const activeShape = shapes.find(
+      shape => shape?.highlightId && activeIds?.includes(shape.highlightId)
+    );
+    if (activeShape) {
+      const timer = setTimeout(() => {
+        const highlightDiv = highlightDivRef.current;
+        if (!highlightDiv) return;
+
+        const highlightElm = highlightDiv?.querySelector(
+          `[data-highlight-id=${activeShape.highlightId}]`
+        );
+        highlightElm?.firstElementChild?.scrollIntoView({ block: 'nearest', inline: 'nearest' });
+      }, 50);
```

Comment:
Why `50` for the timeout? I don't like having arbitrary timeout values, since they may work on your computer, but may fail on a slower computer.

Using zero is fine since that's useful to make some code asynchronous. But it would be better to key off of something else rather than a random timeout, if possible.

Reply:
The intention was to push a new task to the javascript runtime. So I use the value `0`.